src/symmetric_heap_c.c

```diff
@@ -411,8 +411,8 @@ shmemx_malloc_with_hints(size_t size, long hints)
     if (size == 0) return ret;
 
     // Check for valid hints
-    if(hints >= SHMEMX_MALLOC_MAX_HINTS) {
-        RAISE_WARN_STR("%ld is and invalid hint for shmem_malloc_with_hints. Valid hints can be 0, SHMEMX_MALLOC_ATOMICS_REMOTE, SHMEMX_MALLOC_SIGNAL_REMOTE or a combination of these three");
+    if(hints >= SHMEMX_MALLOC_MAX_HINTS || hints < 0) {
+        RAISE_WARN_MSG("%ld is an invalid hint for shmem_malloc_with_hints. Valid hints can be 0, SHMEMX_MALLOC_ATOMICS_REMOTE, SHMEMX_MALLOC_SIGNAL_REMOTE or a combination of these three\n", hints);
```

Comment:
The `_MSG` macros include the `\n`, so no need for it here.

Reply:
I had thought so but when I ran the test it didn't include \n in the output. I'll check again and remove it. Thanks!

Reply:
Actually RAISE_WARN_MSG does not include ' \n' in it which is kind of strange because it should. So I'll keep this '\n' here. There might be some important reason for RAISE_WARN_MSG to not have \n in it so I would not change that.

Reply:
The `_STR` macros include the newline. `_MSG` macros assume you will format the message.

Reply:
My bad - thanks.